### PR TITLE
Native Client port

### PR DESCRIPTION
**Pull request by sbc100**
Hi guys,

This is an initial port of cocos2dx to Native Client (NaCl).

I've tried to follow your coding standards and put you license header at the top of all
new source files.

You can see the port in action (if you are using chrome) here:

https://chrome.google.com/webstore/detail/cocos2dx-test-app/hbnepalailfleigebaanjnoiapedgcpn

### CODE REVIEW

**Comment by dumganhar** [comment]
Hi @sbc100 ,
`This pull request cannot be automatically merged.`
Could you pull our latest code and resolve the conflict ? Thanks.

File: cocos2dx/platform/CCImageCommonWebp.cpp [hunk 1]
```diff
@@ -24,7 +24,13 @@
 
 #include "platform/CCImage.h"
 #include "textures/CCTexture2D.h"
+#ifdef __native_client__
+// TODO(sbc): I'm pretty sure all platforms should be including
+// webph headers in this way.
+#include "webp/decode.h"
+#else
 #include "decode.h"
+#endif
```
Comment (dumganhar) [inline cocos2dx/platform/CCImageCommonWebp.cpp:33]: I think you are right. I will refactor it later. Thanks.

**Comment by sbc100** [comment]
I rebased my changes and fixed merge conflicts.

Also fixed a compile error on CCPhysicsSprite.cpp with box2d support.

Can you try the merge again?

**Comment by minggo** [comment]
Is there any document describes how to build and run native client app?

**Comment by sbc100** [comment]
Sorry I didn't respond to your questions about where the documentation is for how to build and run a NaCl app.

The main docs are here:

https://developers.google.com/native-client/

I've put some cococs2dx specific instructions on how to build in the "build-nacl.sh" script.  However, I think it would be good to have some platform-specific docs in the cocos2dx repo.

Perhaps in the main README?  Or in README.platform_name?

**Comment by walzer** [comment]
Normally we put the documents in http://www.cocos2d-x.org/projects/cocos2d-x/wiki/Setting_up_Development_Environments this wiki page. It's easier to be searched.
And excuse me, I couldn't find the cocos2dx specific instructions on https://developers.google.com/native-client/. Is there any direct link to your document?

**Comment by minggo** [comment]
@sbc100 
Can you give me your email address? I met problems when building for native client.
My email is xiaoming.zhang@cocos2d-x.org.

Thank you.

**Comment by sbc100** [comment]
I added some documentation to the wiki.  Let me know if you have any problems.

**Pull request merged**